src/main/webapp/app/quiz/edit/short-answer-question/edit-short-answer-question.component.ts

```diff
@@ -568,4 +584,115 @@ export class EditShortAnswerQuestionComponent implements OnInit, OnChanges, Afte
        this.questionEditor.getEditor().setValue(this.generateMarkdown());
        this.questionEditor.getEditor().clearSelection();
     }
+
+    /**
+     * For Re-evaluate
+     */
+
+    /**
+     * @function moveUp
+     * @desc Move this question one position up
+     */
+    moveUp() {
+        this.questionMoveUp.emit();
+    }
+
+    /**
+     * @function moveDown
+     * @desc Move this question one position down
+     */
+    moveDown() {
+        this.questionMoveDown.emit();
+    }
+
+    /**
+     * @function
+     * @desc Resets the question title
+     */
+    resetQuestionTitle() {
+        this.question.title = this.backupQuestion.title;
+    }
+
+    /**
+     * @function resetQuestionText
+     * @desc Resets the question text
+     */
+    resetQuestionText() {
+        this.question.text = this.backupQuestion.text;
+        this.question.spots = JSON.parse(JSON.stringify(this.backupQuestion.spots));
+        this.textParts = this.question.text.split(/\n+/g).map(t => t.split(/\s+(?![^[]]*])/g));
```

Comment:
Can you extract a constant for the 2 split regex expressions? I guess you use them several times in your code and in case they change in the future, this would be difficult to maintain. Also make sure to describe it in the comment what the regex actually does

Reply:
I wrote a detailed description in ngOnInit and separated the split (see code below). Is it okay to write the comment like this, or do you want the detailed description everywhere when I use textParts? 

// split on every whitespace. !!!only exception: [-spot 1] is not split!!! for more details see description in ngOnInit.

       const textForEachLine = this.question.text.split(/\n+/g);
       this.textParts = textForEachLine.map(t => t.split(/\s+(?![^[]]*])/g));

Detailed description:
/** We create now the structure on how to display the text of the question
         * 1. The question text is split at every new line. The first element of the array would be then the first line of the question text.
         * 2. Now each line of the question text will be divided into each word (we used whitespace as separator).
         * Note: As the spot tag ( e.g. [-spot 1] ) has in between a whitespace we use regex to not take whitespaces as a separator in between
         * these [ ] brackets.
         * **/